solidity/random-beacon/contracts/RandomBeacon.sol

```diff
@@ -569,6 +646,31 @@ contract RandomBeacon is Ownable {
         }
     }
 
+    /// @notice Punishes the given operators by banning their sortition pool rewards.
+    /// @dev By the way, this function releases gas deposits made by operators
+    ///      during their registration. See `registerOperator` function. This
+    ///      action makes punishments cheaper gas-wise.
+    /// @param ids IDs of punished operators.
+    /// @param punishmentDuration Duration of the punishment period in seconds.
+    function punishOperators(uint32[] memory ids, uint256 punishmentDuration)
+        internal
+    {
+        address[] memory operators = sortitionPool.getIDOperators(ids);
+
+        for (uint256 i = 0; i < operators.length; i++) {
+            // TODO: Do we need the operator to re-deposit gas once
+            //       current punishment is completed to use it for
+            //       future ones?
+            gasStation.releaseGas(operators[i]);
+        }
+
+        // TODO: Once `banRewards` is implemented on the pool side, make sure
+        //       it does not have an unexpected revert instruction which will
+        //       block entry submission. For example, an operator leaves
+        //       the pool just before it gets banned.
```

Comment:
For this reason, I'd wrap the entire block when `punishOperators` and `staking.slash` are called in a try-catch. I don't think we should trust these contracts never revert, especially that staking contract is upgradeable. Not necessarily in this PR though.

Reply:
Done in https://github.com/keep-network/keep-core/pull/2707/commits/eb70b61db31d2351aad51fd88b3ea905449aaaea as part of https://github.com/keep-network/keep-core/pull/2707